lib/c-api/src/wasm_c_api/unstable/middlewares/metering.rs

```diff
@@ -0,0 +1,350 @@
+//!         assert(wasmer_metering_points_is_exhausted(metering_points) == false);
+//!         wasmer_metering_points_delete(metering_points);
+//!     }
+//!
+//!     // Let's call `add_two` for the second time!
+//!     {
+//!         wasm_trap_t* trap = wasm_func_call(add_two, &arguments_as_array, &results_as_array);
+//!         assert(trap == NULL);
+//!         assert(results[0].of.i32 == 42);
+//!
+//!         // There is 2 points left!
+//!         wasmer_metering_points_t* metering_points = wasmer_metering_get_remaining_points(instance);
+//!         assert(wasmer_metering_points_unwrap_or(metering_points, is_exhausted) == 2);
+//!         assert(wasmer_metering_points_is_exhausted(metering_points) == false);
+//!         wasmer_metering_points_delete(metering_points);
+//!     }
+//!
+//!     // Let's call `add_two` for the third time!
+//!     {
+//!         wasm_trap_t* trap = wasm_func_call(add_two, &arguments_as_array, &results_as_array);
+//!         // Oh, it failed!
+//!         assert(trap != NULL);
+//!
+//!         // There is 0 point left… they are exhausted.
+//!         wasmer_metering_points_t* metering_points = wasmer_metering_get_remaining_points(instance);
+//!         assert(wasmer_metering_points_unwrap_or(metering_points, is_exhausted) == is_exhausted);
+//!         assert(wasmer_metering_points_is_exhausted(metering_points) == true);
+//!         wasmer_metering_points_delete(metering_points);
+//!     }
+//!     
+//!     wasm_instance_delete(instance);
+//!     wasm_module_delete(module);
+//!     wasm_store_delete(store);
+//!     wasm_engine_delete(engine);
+//!
+//!     return 0;
+//! }
+//! #    })
+//! #    .success();
+//! # }
+//! ```
+
+use super::super::super::instance::wasm_instance_t;
+use super::super::parser::operator::Operator as COperator;
+use super::wasmer_middleware_t;
+use std::sync::Arc;
+use wasmer::wasmparser::Operator;
+use wasmer_middlewares::{
+    metering::{get_remaining_points, set_remaining_points, MeteringPoints},
+    Metering,
+};
+
+/// Opaque type representing metering points, i.e. the actual number
+/// of remaining points for a given [`wasmer_metering_t`].
+///
+/// To get a value of that type, see the
+/// [`wasmer_metering_get_remaining_points`].
+///
+/// # Example
+///
+/// See module's documentation.
+#[allow(non_camel_case_types)]
+pub struct wasmer_metering_points_t {
+    pub(crate) inner: MeteringPoints,
+}
+
+/// Deletes a [`wasmer_metering_points_t`].
+///
+/// # Example
+///
+/// See module's documentation.
+#[no_mangle]
+pub unsafe extern "C" fn wasmer_metering_points_delete(
+    _metering_points: Option<Box<wasmer_metering_points_t>>,
+) {
+}
+
+/// Returns the number of remaining points if any, otherwise returned
+/// the given `exhausted` value if points are exhausted.
+///
+/// # Example
+///
+/// See module's documentation.
+#[no_mangle]
+pub unsafe extern "C" fn wasmer_metering_points_unwrap_or(
```

Comment:
This is a very Rust-style function (`unwrap_or`). 

Wouldn't it be better if:
* we rename this to something like `wasmer_metering_points_remaining`;
* we hard-code the exhausted value here (i.e `-1`);
* we either return the number or remaining points or `-1`